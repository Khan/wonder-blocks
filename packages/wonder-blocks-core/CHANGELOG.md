# @khanacademy/wonder-blocks-core

## 5.4.0

### Minor Changes

-   1344436f: Add `PropsFor` type to simplify `JSX.LibraryManagedAttributes` usage

## 5.3.1

### Patch Changes

-   9f3752d4: Used named functions in componenets with forwarded refs

## 5.3.0

### Minor Changes

-   c37b99aa: Forward refs for Text and Title
-   c4cef3e6: Added ref forwarding to View

## 5.2.3

### Patch Changes

-   d4c412b5: Update useForceUpdate to ensure consuming hooks properly refresh

## 5.2.2

### Patch Changes

-   64a188e3: Fix useForceUpdate so that regardless of how many times it is called before a new render, it will always cause a new render

## 5.2.1

### Patch Changes

-   5a1ea891: Make sure that React contexts are named
-   df9a10aa: Update state and props to be readonly in components using getDerivedStateFromProps()

## 5.2.0

### Minor Changes

-   fa70c895: Add 'lang', 'className' and 'htmlFor' properties to Text and View

### Patch Changes

-   19ab0408: Update flowgen to convert React.ForwardRefExoticComponent<> and React.FowardedRef<> properly
-   fa70c895: Remove all TypeScript error suppressions on JSX attributes

## 5.1.0

### Minor Changes

-   3c400719: Add useLatestRef hook to wonder-blocks-core

### Patch Changes

-   a6164ed0: Don't use React.FC<> for functional components

## 5.0.4

### Patch Changes

-   c20f48f3: Don't transpile classes when building bundles

## 5.0.3

### Patch Changes

-   b281e2eb: Fix generate Flow types for real

## 5.0.2

### Patch Changes

-   21ce20c7: Fix issues with generate flow types in wonder-blocks-core

## 5.0.1

### Patch Changes

-   ccb6fe00: Miscellaneous TS type fixes
-   d4c2b18c: Fix a variety of issues with Flow types generated by flowgen

## 5.0.0

### Major Changes

-   1ca4d7e3: Fix minor issue with generate Flow types (this is a major bump b/c I forgot to do one after doing the TS conversion)

## 4.9.1

### Patch Changes

-   b5ba5568: Ensure that flow lib defs use React.ElementConfig<> isntead of JSX.LibraryManagedAttributes<>

## 4.9.0

### Minor Changes

-   779b031d: Update AriaProps to use React's AriaAttributes and AriaRole types

## 4.8.0

### Minor Changes

-   873f4a14: Update aphrodite lib def and StyleType

### Patch Changes

-   d816af08: Update build and test configs use TypeScript
-   3891f544: Update babel config to include plugins that Storybook needed
-   0d28bb1c: Configured TypeScript
-   3d05f764: Fix HOCs and other type errors
-   c2ec4902: Update eslint configuration, fix lint
-   2983c05b: Include 'types' field in package.json
-   77ff6a66: Generate Flow types from TypeScript types
-   ec8d4b7f: Fix miscellaneous TypeScript errors

## 4.7.0

### Minor Changes

-   91cb727c: Convert enums to POJOs

### Patch Changes

-   91cb727c: Remove file extensions from imports

## 4.6.2

### Patch Changes

-   496119f2: Cleanup WB interdependencies

## 4.6.1

## 4.6.0

### Minor Changes

-   b561425a: Add useIsMounted() hook
-   a566e232: Add useOnMountEffect hook
-   d2b21a6e: Export useOnMountEffect hook

## 4.5.0

### Minor Changes

-   175a2dd2: Add 'useRenderState' hook

## 4.4.0

### Minor Changes

-   6ee20af9: Add `useOnline` hook to encapsulate navigator.onLine and the offline/online events

## 4.3.2

### Patch Changes

-   5f4a4297: Make dependency on `flow-enums-runtime` explicit
-   2b96fd59: Change flow-enums-runtime to be peer dependencies

## 4.3.1

## 4.3.0

### Minor Changes

-   246a921d: NEW: `useForceUpdate` hook. This should rarely be used and likely only ever from other hooks.

## 4.2.1

## 4.2.0

### Minor Changes

-   0aed8723: Adds `throwIfNested` required prop to `RenderStateRoot`.

## 4.1.0

### Minor Changes

-   45588e5f: Fix an issue with `useUniqueIdWithMock`/`useUniqueIdWithoutMock`
    rerender more than was needed. The fix introduces `<RenderStateRoot>`
    which must be an ancestor to all components uses these hooks.
-   875b7893: Nesting of `RenderStateRoot`s inside each other can result in extra renders
    and potentially incorrect behavior. `RenderStateRoot` now throws if it
    appears as a descendent of another `RenderStateRoot`.
