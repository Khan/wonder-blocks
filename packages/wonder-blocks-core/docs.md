`wonder-blocks-core` provides building blocks for other `wonder-blocks` 
components.  These components, `View` and `Text` can also be used on their 
own.

`View` roughly maps to `div` and `Text` roughly maps to `span`.

```js
const {StyleSheet} = require("aphrodite");

const styles = StyleSheet.create({
    container: {
        padding: 32,
        backgroundColor: "plum",
    },
    text: {
        fontFamily: "sans-serif",
        fontSize: 24,
    },
});

<View style={styles.container}>
    <Text style={styles.text}>
        Hello, world!
    </Text>
</View>
```

Other props can be passed through `View` or `Text`, as if they were normal tags.

```js
<View>
    <View onClick={() => alert("Clicked!")}>
        Click me!
    </View>

    <Text aria-hidden>
        This text is hidden from screen readers.
    </Text>
</View>
```

There's also an HOC that adds the same style prop to existing 
components.

```js
const {addStyle} = require("./index.js");
const {StyleSheet} = require("aphrodite");

const styles = StyleSheet.create({
    // default style for all instances of StyledInput
    input: {
        fontSize: 30,
    },
    // style for a particular instance of StyledInput
    pink: {
        backgroundColor: "pink",
    },
});

const StyledInput = addStyle("input", styles.input);

<StyledInput style={styles.pink} type="text" placeholder="hello, world"/>;
```
