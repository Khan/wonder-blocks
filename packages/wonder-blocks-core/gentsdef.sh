#!/usr/bin/env bash
# TODO: figure out why writing this as script in package.json doesn't work
# My hunch is that it has to do with where tsconfig.json is located
tsc --emitDeclarationOnly --declaration --declarationDir dist --allowSyntheticDefaultImports --downlevelIteration --jsx react --lib ES2018,DOM index.ts
