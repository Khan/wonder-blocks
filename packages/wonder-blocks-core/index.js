// @flow
import type {
    ClickableHandlers,
    ClickableState,
} from "./components/clickable-behavior.js";
import type {AriaProps, IIdentifierFactory, StyleType} from "./util/types.js";
import type {Intersection} from "./util/get-element-intersection.js";

export {default as ClickableBehavior} from "./components/clickable-behavior.js";
export {default as Text} from "./components/text.js";
export {default as View} from "./components/view.js";
export {
    default as WithSSRPlaceholder,
} from "./components/with-ssr-placeholder.js";
export {default as IDProvider} from "./components/id-provider.js";
export {default as UniqueIDProvider} from "./components/unique-id-provider.js";
export {default as addStyle} from "./util/add-style.js";
export {
    default as getClickableBehavior,
} from "./util/get-clickable-behavior.js";
export {
    default as getElementIntersection,
} from "./util/get-element-intersection.js";

export type {
    AriaProps,
    ClickableHandlers,
    ClickableState,
    Intersection,
    IIdentifierFactory,
    StyleType,
};
