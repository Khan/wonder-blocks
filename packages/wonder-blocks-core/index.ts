// @flow
import {ClickableHandlers} from "./components/clickable-behavior";
import {AriaProps, IIdentifierFactory, StyleType} from "./util/types";
import {Intersection} from "./util/get-element-intersection";

export {default as ClickableBehavior} from "./components/clickable-behavior";
export {default as Text} from "./components/text";
export {default as View} from "./components/view";
export {
    default as WithSSRPlaceholder,
} from "./components/with-ssr-placeholder";
export {default as UniqueIDProvider} from "./components/unique-id-provider";
export {default as addStyle} from "./util/add-style";
export {
    default as getClickableBehavior,
} from "./util/get-clickable-behavior";
export {
    default as getElementIntersection,
} from "./util/get-element-intersection";

export {
    AriaProps,
    ClickableHandlers,
    Intersection,
    IIdentifierFactory,
    StyleType,
};
