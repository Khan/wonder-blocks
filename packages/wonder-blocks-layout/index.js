// @flow
import type {MediaSize, MediaSpec} from "./util/types.js";

export {default as MediaLayout} from "./components/media-layout.js";
export {default as Layout} from "./components/layout.js";
export {default as Spring} from "./components/spring.js";
export {default as Strut} from "./components/strut.js";
export * from "./util/specs.js";
export {matchesSize} from "./util/util.js";

export type {MediaSize, MediaSpec};
