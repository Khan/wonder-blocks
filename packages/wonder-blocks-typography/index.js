import Title from "./src/title.js";
import HeadingLarge from "./src/heading-large.js";
import LabelLarge from "./src/label-large.js";

const Heading = {
    Large: HeadingLarge,
};

const Label = {
    Large: LabelLarge,
};

export {Title, Heading, Label};
