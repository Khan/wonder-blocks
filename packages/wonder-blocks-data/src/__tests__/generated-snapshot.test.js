// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-data
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");
import {Body, BodyMonospace} from "@khanacademy/wonder-blocks-typography";
import {View, Server} from "@khanacademy/wonder-blocks-core";
import {
    Data,
    RequestHandler,
    initializeCache,
    InterceptData,
    TrackData,
    fulfillAllDataRequests,
} from "@khanacademy/wonder-blocks-data";
import {Strut} from "@khanacademy/wonder-blocks-layout";
import Color from "@khanacademy/wonder-blocks-color";
import Spacing from "@khanacademy/wonder-blocks-spacing";
import Button from "@khanacademy/wonder-blocks-button";

describe("wonder-blocks-data", () => {
    it("example 1", () => {
        class MyValidHandler extends RequestHandler {
            constructor() {
                super("CACHE_MISS_HANDLER_VALID");
            }

            fulfillRequest(options) {
                return new Promise((resolve, reject) =>
                    setTimeout(() => resolve("I'm DATA from a request"), 3000),
                );
            }
        }

        class MyInvalidHandler extends RequestHandler {
            constructor() {
                super("CACHE_MISS_HANDLER_ERROR");
            }

            fulfillRequest(options) {
                return new Promise((resolve, reject) =>
                    setTimeout(
                        () => reject("I'm an ERROR from a request"),
                        3000,
                    ),
                );
            }
        }

        const valid = new MyValidHandler();
        const invalid = new MyInvalidHandler();
        const example = (
            <View>
                <View>
                    <Body>This request will succeed and give us data!</Body>
                    <Data
                        handler={valid}
                        options={{
                            some: "options",
                        }}
                    >
                        {({loading, data}) => {
                            if (loading) {
                                return "Loading...";
                            }

                            return <BodyMonospace>{data}</BodyMonospace>;
                        }}
                    </Data>
                </View>
                <Strut size={Spacing.small_12} />
                <View>
                    <Body>This request will go boom and give us an error!</Body>
                    <Data
                        handler={invalid}
                        options={{
                            some: "options",
                        }}
                    >
                        {({loading, error}) => {
                            if (loading) {
                                return "Loading...";
                            }

                            return (
                                <BodyMonospace
                                    style={{
                                        color: Color.red,
                                    }}
                                >
                                    ERROR: {error}
                                </BodyMonospace>
                            );
                        }}
                    </Data>
                </View>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("example 2", () => {
        class MyHandler extends RequestHandler {
            constructor() {
                super("CACHE_HIT_HANDLER");
            }
            /**
             * fulfillRequest should not get called as we already have data cached.
             */

            fulfillRequest(options) {
                throw new Error(
                    "If you're seeing this error, the examples are broken and data isn't in the cache that should be.",
                );
            }
        }

        const handler = new MyHandler();
        initializeCache({
            CACHE_HIT_HANDLER: {
                DATA: {
                    data: "I'm DATA from the hydration cache",
                },
                ERROR: {
                    error: "I'm an ERROR from hydration cache",
                },
            },
        });
        const example = (
            <View>
                <View>
                    <Body>This cache has data!</Body>
                    <Data handler={handler} options={"DATA"}>
                        {({loading, data}) => {
                            if (loading) {
                                return "If you see this, the example is broken!";
                            }

                            return <BodyMonospace>{data}</BodyMonospace>;
                        }}
                    </Data>
                </View>
                <Strut size={Spacing.small_12} />
                <View>
                    <Body>This cache has error!</Body>
                    <Data handler={handler} options={"ERROR"}>
                        {({loading, error}) => {
                            if (loading) {
                                return "If you see this, the example is broken!";
                            }

                            return (
                                <BodyMonospace
                                    style={{
                                        color: Color.red,
                                    }}
                                >
                                    ERROR: {error}
                                </BodyMonospace>
                            );
                        }}
                    </Data>
                </View>
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("example 3", () => {
        class MyHandler extends RequestHandler {
            constructor() {
                super("INTERCEPT_DATA_HANDLER1");
            }

            fulfillRequest(options) {
                return Promise.reject(new Error("You should not see this!"));
            }
        }

        const handler = new MyHandler();

        const fulfillRequestInterceptor = function (options) {
            if (options === "DATA") {
                return Promise.resolve("INTERCEPTED DATA!");
            }

            return null;
        };

        const example = (
            <InterceptData
                handler={handler}
                fulfillRequest={fulfillRequestInterceptor}
            >
                <View>
                    <Body>This received intercepted data!</Body>
                    <Data handler={handler} options={"DATA"}>
                        {({loading, data}) => {
                            if (loading) {
                                return "If you see this, the example is broken!";
                            }

                            return <BodyMonospace>{data}</BodyMonospace>;
                        }}
                    </Data>
                </View>
            </InterceptData>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("example 4", () => {
        class ErrorBoundary extends React.Component {
            constructor(props) {
                super(props);
                this.state = {};
            }

            static getDerivedStateFromError(error) {
                return {
                    error: error.message,
                };
            }

            render() {
                if (typeof jest !== "undefined") {
                    /**
                     * The snapshot test just sees the error getting thrown, not the
                     * awesome error boundary, so we have to hack around it to keep
                     * this live example, but not get test failures.
                     */
                    return "Sorry, no snapshot for you";
                }

                if (this.state.error) {
                    return <View>{this.state.error}</View>;
                }

                return this.props.children;
            }
        }

        const example = (
            <ErrorBoundary>
                <View>
                    <TrackData>
                        <Body>
                            This only renders if we're in server-side mode and
                            the page hot reloaded
                        </Body>
                    </TrackData>
                </View>
            </ErrorBoundary>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("example 5", () => {
        class MyPretendHandler extends RequestHandler {
            constructor() {
                super("MY_PRETEND_HANDLER");
            }

            fulfillRequest(options) {
                return new Promise((resolve, reject) =>
                    setTimeout(() => resolve("DATA!"), 3000),
                );
            }
        }

        class Example extends React.Component {
            constructor() {
                super();
                /**
                 * For this demonstration, we need to hack the return of isServerSide solely
                 * for the scope of this component.
                 */

                this.state = {};
                this._handler = new MyPretendHandler();
            }

            static getDerivedStateFromError(error) {
                return {
                    error,
                };
            }

            componentDidMount() {
                this._mounted = true;
            }

            componentWillUnmount() {
                this._mounted = false;
            }

            setClientMode() {
                window.location.reload();
            }

            setServerMode() {
                Server.setServerSide();
                this.setState({
                    refresh: Date.now(),
                    error: null,
                });
            }

            _renderErrorOrContent() {
                if (typeof jest !== "undefined") {
                    /**
                     * The snapshot test just sees the error getting thrown, not the
                     * awesome error boundary, so we have to hack around it to keep
                     * this live example, but not get test failures.
                     */
                    return "Sorry, no snapshot for you";
                }

                if (this.state.error) {
                    return (
                        <React.Fragment>
                            <Strut size={Spacing.small_12} />
                            <Body>
                                We can't show you anything useful in client-side
                                mode
                            </Body>
                        </React.Fragment>
                    );
                }

                const data = this.state.data
                    ? JSON.stringify(this.state.data, undefined, "  ")
                    : "Data requested...";
                return (
                    <React.Fragment>
                        <Strut size={Spacing.small_12} />
                        <TrackData>
                            <Data handler={this._handler} options={{}}>
                                {({loading, data, error}) => (
                                    <View>
                                        <BodyMonospace>{`Loading: ${loading}`}</BodyMonospace>
                                        <BodyMonospace>{`Data: ${JSON.stringify(
                                            data,
                                        )}`}</BodyMonospace>
                                    </View>
                                )}
                            </Data>
                        </TrackData>
                        <Strut size={Spacing.small_12} />
                        <View>
                            <Body>
                                The above components requested data, but we're
                                server-side, so all that happened is we tracked
                                the request. In this example, we've also called
                                `fulfillAllDataRequests` to fetch that tracked
                                data.
                            </Body>
                            <Strut size={Spacing.small_12} />
                            <Body>
                                In about 3 seconds, it will appear below. Notice
                                that when it does, the above still doesn't
                                update. That's because during SSR, the data is
                                not updated in the rendered tree.
                            </Body>
                            <Strut size={Spacing.small_12} />
                            <Body>
                                If you click to remount after the data appears,
                                we'll rerender with the now cached data, and
                                above should update accordingly.
                            </Body>
                            <Strut size={Spacing.small_12} />
                            <BodyMonospace>{data}</BodyMonospace>
                        </View>
                    </React.Fragment>
                );
            }

            render() {
                try {
                    return (
                        <View key={this.state.refresh}>
                            {Server.isServerSide() ? (
                                <React.Fragment>
                                    <Button
                                        kind={"secondary"}
                                        onClick={() => this.setClientMode()}
                                    >
                                        Back to Client-side Mode (reloads page)
                                    </Button>
                                    <Strut size={Spacing.small_12} />
                                    <Button
                                        kind={"secondary"}
                                        onClick={() => this.setServerMode()}
                                    >
                                        Re-mount
                                    </Button>
                                </React.Fragment>
                            ) : (
                                <Button
                                    kind={"primary"}
                                    onClick={() => this.setServerMode()}
                                >
                                    Enable Server-side Mode
                                </Button>
                            )}
                            {this._renderErrorOrContent()}
                        </View>
                    );
                } finally {
                    if (!this.state.data && Server.isServerSide()) {
                        setTimeout(
                            () =>
                                fulfillAllDataRequests().then((data) => {
                                    if (this._mounted) {
                                        this.setState({
                                            data,
                                        });
                                    }
                                }),
                            0,
                        );
                    }
                }
            }
        }

        const example = <Example />;
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
