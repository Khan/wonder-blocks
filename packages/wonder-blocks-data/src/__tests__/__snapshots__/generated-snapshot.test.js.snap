// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-data example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      This request will succeed and give us data!
    </span>
    Loading...
  </div>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      aria-hidden="true"
      className=""
      style={
        Object {
          "MsFlexBasis": 12,
          "MsFlexPreferredSize": 12,
          "WebkitFlexBasis": 12,
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": 12,
          "flexDirection": "column",
          "flexShrink": 0,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 12,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      This request will go boom and give us an error!
    </span>
    Loading...
  </div>
</div>
`;

exports[`wonder-blocks-data example 2 1`] = `
Array [
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      This cache has data!
    </span>
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      I'm DATA from the cache
    </span>
  </div>,
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      aria-hidden="true"
      className=""
      style={
        Object {
          "MsFlexBasis": 12,
          "MsFlexPreferredSize": 12,
          "WebkitFlexBasis": 12,
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": 12,
          "flexDirection": "column",
          "flexShrink": 0,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 12,
          "zIndex": 0,
        }
      }
    />
  </div>,
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      This cache has error!
    </span>
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "color": "#d92916",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      ERROR: 
      I'm an ERROR from the cache
    </span>
  </div>,
]
`;

exports[`wonder-blocks-data example 3 1`] = `
Array [
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      This intercepted cache has data!
    </span>
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      I'm DATA from the cache interceptor
    </span>
  </div>,
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      aria-hidden="true"
      className=""
      style={
        Object {
          "MsFlexBasis": 12,
          "MsFlexPreferredSize": 12,
          "WebkitFlexBasis": 12,
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": 12,
          "flexDirection": "column",
          "flexShrink": 0,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 12,
          "zIndex": 0,
        }
      }
    />
  </div>,
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "display": "block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      This intercepted cache has error!
    </span>
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "color": "#d92916",
          "display": "block",
          "fontFamily": "Inconsolata, monospace",
          "fontSize": 17,
          "fontWeight": 400,
          "lineHeight": "22px",
        }
      }
    >
      ERROR: 
      I'm an ERROR from the cache interceptor
    </span>
  </div>,
]
`;

exports[`wonder-blocks-data example 4 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    This received intercepted data!
  </span>
  If you see this, the example is broken!
</div>
`;

exports[`wonder-blocks-data example 5 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <span
    className=""
    style={
      Object {
        "MozOsxFontSmoothing": "grayscale",
        "WebkitFontSmoothing": "antialiased",
        "display": "block",
        "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
        "fontSize": 16,
        "fontWeight": 400,
        "lineHeight": "22px",
      }
    }
  >
    This re-renders once a second. On the render, the cache is refreshed and so we see an update.
  </span>
  If you see this, the example is broken!
</div>
`;

exports[`wonder-blocks-data example 6 1`] = `"Sorry, no snapshot for you"`;

exports[`wonder-blocks-data example 7 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <button
    className=""
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragStart={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    role="button"
    style={
      Object {
        "::MozFocusInner": Object {
          "border": 0,
        },
        ":focus": Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        },
        "alignItems": "center",
        "background": "#1865f2",
        "border": "none",
        "borderRadius": 4,
        "boxSizing": "border-box",
        "color": "#ffffff",
        "cursor": "pointer",
        "display": "inline-flex",
        "height": 40,
        "justifyContent": "center",
        "margin": 0,
        "outline": "none",
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "position": "relative",
        "textDecoration": "none",
        "touchAction": "manipulation",
        "userSelect": "none",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className=""
      style={
        Object {
          "MozOsxFontSmoothing": "grayscale",
          "WebkitFontSmoothing": "antialiased",
          "alignItems": "center",
          "display": "inline-block",
          "fontFamily": "Lato, \\"Noto Sans\\", sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": "20px",
          "overflow": "hidden",
          "pointerEvents": "none",
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      Enable Server-side Mode
    </span>
  </button>
  Sorry, no snapshot for you
</div>
`;
